### PR TITLE
cmake

### PR DESCRIPTION
**Pull request by sunflower-knight**
## Why
1. more easy for developer who use cmake-gui 
2. add more option for developer  like  "Warning for all project or a single target" "ASAN for all project or a single target"
3. manager a lot of cmake in different file like "build.cmake" "install.cmake" . It will be uesfule for future develop .for example if you want to add http module , you need to only change module.cmake. And you should turn on warning or ASAN for http-module only.  

## What is changing
1. for user 
**cmake-build api**  has changed  if you want to compile some of the all project in CLI. You shuld use -Dcore_rpc=ON Dstruct_pack=ON . And you can use cmake-gui more easyily  .....
2.  for developre 
add some function for developers and  cmake become easyier to read 

## Example
image

### CODE REVIEW

**Comment by PikachuHyA** [comment]
please sign CLA

File: cmake/build.cmake [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Compile Standard
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+message("Cxx Standard: ${CMAKE_CXX_STANDARD}")
```
Comment (PikachuHyA) [inline cmake/build.cmake:4]: change all message("str") to message(STATUS "str")

File: CMakeLists.txt [hunk 2]
```diff
@@ -5,107 +5,13 @@ project(yaLanTingLibs
...
-endforeach ()
 
-install(DIRECTORY include DESTINATION include)
+include(cmake/utils.cmake)
+include(cmake/build.cmake)
+include(cmake/platform.cmake)
+include(cmake/dependency.cmake)
+include(cmake/develop.cmake)
+include(cmake/install.cmake)
+include(cmake/module.cmake)
```
Comment (PikachuHyA) [inline CMakeLists.txt:17]: add extra empty line
keep last line empty

check other files too

**Comment by sunflower-knight** [comment]
I am trying to fix some bug and sign CLA

File: cmake/develop.cmake [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+endif ()
+
+# Enable address sanitizer
+option(ENABLE_SANITIZER "Enable sanitizer(debug+gcc/clang/appleclang)" ON)
+if (ENABLE_SANITIZER)
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        if (NOT CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang|AppleClang")
+            message(FATAL_ERROR "Sanitizer supported only for Gcc/Clang/AppleClang")
+        endif ()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
```
Comment (PikachuHyA) [inline cmake/develop.cmake:32]: before use `-fsanitize=address`,  check the flag with `CHeckASan.cmake` is necessary.
due to corner case, the CI will fail.
e.g. the lib `sanitize` is not found, while the compiler work well

**Review by PikachuHyA — CHANGES_REQUESTED** [review]

**Comment by sunflower-knight** [comment]
fix all problem mentioned above

**Pull request closed without merging**